packages/compass-sidebar/src/components/multiple-connections/saved-connections/saved-connection.tsx

```diff
@@ -0,0 +1,260 @@
+import {
+  type ConnectionInfo,
+  getConnectionTitle,
+} from '@mongodb-js/connection-info';
+import React, { useCallback } from 'react';
+import {
+  css,
+  spacing,
+  Icon,
+  IconButton,
+  ItemActionControls,
+  useHoverState,
+  useToast,
+} from '@mongodb-js/compass-components';
+import type { ItemAction } from '@mongodb-js/compass-components';
+import { useConnectionColor } from '@mongodb-js/connection-form';
+import { useMaybeProtectConnectionString } from '@mongodb-js/compass-maybe-protect-connection-string';
+import type { ItemSeparator } from '@mongodb-js/compass-components/lib/components/item-action-controls';
+
+const TOAST_TIMEOUT_MS = 5000; // 5 seconds.
+
+const iconStyles = css({
+  fontSize: 14,
+  minWidth: '14px',
+  maxWidth: '14px',
+  minHeight: '14px',
+  maxHeight: '14px',
```

Comment:
`Icon` components also have `size` property that you might want to use instead of this maybe

Reply:
The design is explicitly saying 14px and I don't see a spacing token of 14px in leafygreen:

https://www.mongodb.design/component/tokens/example/

What I can do is to use the spacing of 16px we have there, it will be 2px bigger, but I don't think that would be a big issue. WDYT?

Reply:
That works for the icons, now I'll need to create a custom icon for the SVG so I can use small there too. Trying it.